src/saltext/vmware/modules/vmc_org_users.py

```diff
@@ -0,0 +1,441 @@
+    )
+    return vmc_request.call_api(
+        method=vmc_constants.GET_REQUEST_METHOD,
+        url=api_url,
+        refresh_key=refresh_key,
+        authorization_host=hostname,
+        description="vmc_org_users.list",
+        params=params,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+
+def search(
+    hostname, refresh_key, org_id, user_search_term, expand_profile=None, verify_ssl=True, cert=None
+):
+    """
+    Search users in organization having username, firstName, lastName or email which "contains" search term.
+    e.g. search for "test" will return test@vmware.com if test@vmware.com is part of the organization.
+    Search results limited to first 20 results. Please refine the search term for accurate results.
+    Organization members will receive basic user information.
+    Organization owners and Service Owners (for organizations that have access to the service) will additionally receive role details of the users.
+
+    Please refer the `Search Org Users documentation <https://developer.vmware.com/apis/csp/csp-iam/latest/csp/gateway/am/api/orgs/orgId/users/search/get/>`_ to get insight of functionality and input parameters
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt <minion id> vmc_org_users.search hostname=console.cloud.vmware.com org_id=org_id  ...
+
+    hostname
+        The host name of CSP
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    org_id
+        The Id of organization for which user list is retrieved
+
+    user_search_term: String
+        (Required) The string to be searched within email or firstName or lastName or username
+
+    expand_profile: String
+        (Optional) Indicates if the response should be expanded with the user profile, the value is ignored, only the existence of parameter is checked.
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    """
+
+    log.info("Searching for Org users in the  org %s", org_id)
+    api_base_url = vmc_request.set_base_url(hostname)
+    api_url = "{base_url}csp/gateway/am/api/orgs/{org_id}/users/search".format(
+        base_url=api_base_url, org_id=org_id
+    )
+
+    params = vmc_request._filter_kwargs(
+        allowed_kwargs=["userSearchTerm", "expandProfile"],
+        userSearchTerm=user_search_term,
+        expandProfile=expand_profile,
+    )
+
+    return vmc_request.call_api(
+        method=vmc_constants.GET_REQUEST_METHOD,
+        url=api_url,
+        refresh_key=refresh_key,
+        authorization_host=hostname,
+        description="vmc_org_users.search",
+        params=params,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+
+def add(
```

Comment:
**question** how is this function supposed to work? I'm trying to add a user and it returns successfully, but the user doesn't show up in the `list` or `search` functionality :thinking:

Reply:
As we can not expose vmware specific data, we are trying to add a testuser.
The underlying invite API checks if the passed user is part of vmware Active Directory or not. If the user is part of it, then only the underlying invite API will add the user to the given org.

As the testuser is some dummy value and is not part of the vmware Active Directory, the invite API just sends 200 OK response though it did not added the user to the org.

so we are testing the flow as:
- add the user (as the user is not part of vmware Active Directory, the underlying API just sends 200 OK response)
- so when we call search function by  passing the search term as testuser, it should return empty result

Reply:
Sounds like the VMC can be configured, I'm guessing, for different backing LDAP (or other?) account stores. Obviously we don't have any control over that and just have to work within the limits of the API... I think that just calling it `invite` is enough here... even if I'm not happy about it :joy: